validator/lib/validateCmpt.js

```diff
@@ -0,0 +1,136 @@
+        };
+    }
+
+    if (version !== 1) {
+        return {
+            result : false,
+            message: 'Cmpt header has an invalid version field. Expected version = 1. Found version = ' + version
+        };
+    }
+
+    if (byteLength !== content.length) {
+        return {
+            result : false,
+            message: 'Cmpt header has an invalid byteLength field. Expected byteLength = ' + content.length + '. Found byteLength = ' + byteLength
+        };
+    }
+
+    for (var i = 0; i < tilesLength; i++) {
+        var errorAddon = '';
+        var innerTileMagic = content.toString('utf8', byteOffset, byteOffset + sizeOfUint32);
+        var innerTileByteLength = content.readUInt32LE(byteOffset + 2 * sizeOfUint32);
+
+        var innerTile;
+        var validatorResult;
+
+        if (innerTileMagic === 'b3dm') {
+            innerTile = new Buffer(24);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateB3dm(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'i3dm'){
+            innerTile = new Buffer(32);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateI3dm(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'pnts'){
+            innerTile = new Buffer(28);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validatePnts(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else if (innerTileMagic === 'cmpt') {
+            innerTile = new Buffer(innerTileByteLength);
+            content.copy(innerTile, 0, byteOffset, byteOffset + innerTileByteLength);
+            validatorResult = validateCmpt(innerTile);
+            isValid = isValid && validatorResult.result;
+        }
+
+        else {
+            isValid = false;
+            errorAddon += 'Inner tile header magic cannot be identified';
+        }
+
+        byteOffset = byteOffset + innerTileByteLength; // skip over this tile
+
+        if( (byteOffset + 12 > byteLength) && (i + 1 !== tilesLength) ) {
+            isValid = false;
+            errorAddon += 'Cmpt header given tilesLength = ' + tilesLength + '. Found number of inner tiles = ' + (i + 1);
+        }
+
+        if (!isValid) {
+            var errorMessage = 'Cmpt header has an invalid inner tile:\n';
+            errorMessage += 'Invalid inner tile index = ' + i + ' starting at byte = ' + (byteOffset - innerTileByteLength) + '\n';
+            errorMessage += 'Invalid inner tile has magic = ' + innerTileMagic;
+            if (defined(validatorResult)) {
+                errorMessage += '\n' + validatorResult.message;
+            }
+
+            if(errorAddon.length !== 0) {
```

Comment:
`if (`

Instead of having both `validatorResult.message` and `errorAddon` you can combine these into the same variable. Then the code becomes shorter and simpler:

```javascript
errorMessage += 'Invalid inner tile has magic = ' + innerTileMagic + '\n';
errorMessage += errorAddon;
```